tools/chef/roles/db-mysql-upgrade.json

```diff
@@ -0,0 +1,62 @@
+{
+  "name": "db-mysql-upgrade",
+  "chef_type": "role",
+  "json_class": "Chef::Role",
+  "description": "MySQL upgrader. Be careful!",
+  "default_attributes": {
+    "mysql-libs": {
+      "replace_packages": [
+        "mysql-libs",
+        "mysql55w-libs",
+        "libmysqlclient16",
+        "mariadb-libs",
+        "Percona-Server-shared-55"
+      ]
+    },
+    "mysql-client": {
+      "replace_packages": [
+        "mysql",
+        "mariadb",
+        "Percona-Server-client-51",
+        "Percona-Server-client-55"
+      ]
+    },
+    "mysql": {
+      "replace_packages": [
+        "mysql-server",
+        "mariadb-server",
+        "Percona-Server-server-51",
+        "Percona-Server-server-55"
+      ]
+    },
+    "package_replacements": {
+      "mysql-libs": {
+        "enabled": true,
+        "strategy": "yum_shell",
+        "from": "replace_packages",
+        "to": "replace_package_targets",
```

Comment:
The idea was to have it re-use things like `"php": {"packages": [...]}` without having to specify each one, but that got put aside once we determined that yum replace only handles one argument.

So, we can massively simplify the syntax!